templates/serverpod_templates/projectname_server_upgrade/deploy/aws/terraform/code-deploy.tf

```diff
@@ -89,3 +89,10 @@ resource "aws_s3_bucket_acl" "deployment" {
   bucket = aws_s3_bucket.deployment.id
   acl    = "private"
 }
+
+resource "aws_s3_bucket_ownership_controls" "deployment" {
+  bucket = aws_s3_bucket.deployment.id
+  rule {
+    object_ownership = "ObjectWriter"
+  }
+}
```

Comment:
Question: Just for my understanding, what does this do exactly?

Reply:
aws s3 bucket by default have ACL disabled. To enable ACL, we need to set the controlling ownership.

there are 2 types of ownership:
`BucketOwnerPreferred` allow the writter to transfer ownership to bucket owner, and `ObjectWritter` will own the object but can grant object access permissions to others.  :)

Reply:
further reads:
when they introduced it: https://aws.amazon.com/about-aws/whats-new/2021/11/amazon-s3-object-ownership-simplify-access-management-data-s3/
terraform explanation: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_ownership_controls
aws explanation:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/about-object-ownership.html